### PR TITLE
New version of xcat test main script in jenkins environment

### PR DESCRIPTION
**Pull request by hu-weihua**
There are too many commits in old pull request #900 to cover new requirements , it isn't convenient to review. So close old #900, and open this new one.

### CODE REVIEW

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 1]
```diff
@@ -0,0 +1,1312 @@
...
+            &runcmd("cd $miniisopath >/dev/null 2>&1 && wget $downloadpath >/dev/null 2>&1");
+            if($?){
+                send_msg(0, "[mn_install] Can't download mini.iso from $downloadpath");
+                return 1;
+            }
+            send_msg(2, "[mn_install] download mini.iso from $downloadpath  ....[done]");
+            rename("/install/$os/ppc64el/install/netboot/initrd.gz", "/install/$os/ppc64el/install/netboot/initrd.gz.bakupby$proid") if(-e "/install/$os/ppc64el/install/netboot/initrd.gz"); 
+            mkpath("$miniisopath/mountpoint");
+            &runcmd("mount -o loop $miniisopath/mini.iso $miniisopath/mountpoint");
+            &runcmd("cp $miniisopath/mountpoint/install/initrd.gz /install/$os/ppc64el/install/netboot");
```
Comment (tingtli) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: Need to first mkdir -p /install/$os/ppc64el/install/netboot before copy the initrd.gz to it

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 2]
```diff
@@ -0,0 +1,1312 @@
...
+        $os =~ /(\D+)(\d+)\.?(\d?)/;       
+        &runcmd("xdsh $mn \"cd /xcat-core && ./mklocalrepo.sh\" >/dev/null 2>&1");
+        my $version=$2;
+        if($arch =~ /ppc64le/i || $arch =~ /ppc64el/i){
+            &runcmd("xdsh $mn \"cd /xcat-dep/rh$version/ppc64le && ./mklocalrepo.sh\" >/dev/null 2>&1");
+        }else{
+            &runcmd("xdsh $mn \"cd /xcat-dep/rh$version/$arch && ./mklocalrepo.sh\" >/dev/null 2>&1");
+        }
+        &runcmd("xdsh $mn \"rpm --import /RPM-GPG-KEY-redhat-release\" >/dev/null 2>&1");
+        &runcmd("xdsh $mn \"yum -y install xCAT xCAT-test > $installlog 2>&1\" >/dev/null 2>&1");
```
Comment (tingtli) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: It is xcat-test , not xCAT-test

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 3]
```diff
@@ -0,0 +1,1313 @@
...
+}else{
+    send_msg(0, "[[main]]: regression test return out of time");
+    if($pid) {
+        my $try=1;
+        do{
+            kill 'INT', $pid;
+            send_msg(2, "[[main]]: send INT to subprocess $pid...[$try]");
+            $try++;
+            sleep 1;
+        }while(waitpid($pid,  WNOHANG)==0)
```
Comment (neo954) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: I don't think keep sending ``SIGINT`` is a good method. I suggest if a couple of trying to send ``SIGINT`` is not working, sending ``SIGTERM``, and if this is still not working, sending ``SIGKILL``.

In this way, we can make sure child process is killed.

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest (hunk 2 above)
Comment (neo954) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: @tingtli, It is ``xCAT-test`` for RHEL and SLES, and ``xcat-test`` for Debian/Ubuntu.

Comment (tingtli) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: oh, yes!

**Comment by tingtli** [comment]
@neo954  i think the code is good for me. I would like to merge it. Do you have further comments?

**Comment by neo954** [comment]
Please merge it, @tingtli

**Pull request merged**